specutils/tests/test_loaders.py

```diff
@@ -117,34 +122,64 @@ def test_hst_stis(remote_data_path):
     assert isinstance(spec, Spectrum1D)
     assert spec.flux.size > 0
 
+    # HDUList case
+    hdulist = fits.open(remote_data_path)
+    spec = Spectrum1D.read(hdulist, format="HST/STIS")
+    assert isinstance(spec, Spectrum1D)
+    assert spec.flux.size > 0
 
 @pytest.mark.remote_data
 def test_sdss_spec():
     sp_pattern = 'spec-4055-55359-0596.fits.'
+    with urllib.request.urlopen('https://dr14.sdss.org/optical/spectrum/view/data/format%3Dfits/spec%3Dlite?mjd=55359&fiberid=596&plateid=4055') as response:
+        # Read from open file object
+        spec = Spectrum1D.read(response, format="SDSS-III/IV spec")
+        assert isinstance(spec, Spectrum1D)
+        assert spec.flux.size > 0
+
     with urllib.request.urlopen('https://dr14.sdss.org/optical/spectrum/view/data/format%3Dfits/spec%3Dlite?mjd=55359&fiberid=596&plateid=4055') as response:
         with tempfile.NamedTemporaryFile(prefix=sp_pattern) as tmp_file:
             shutil.copyfileobj(response, tmp_file)
 
+            # Read from filename
             spec = Spectrum1D.read(tmp_file.name)
 
             assert isinstance(spec, Spectrum1D)
             assert spec.flux.size > 0
 
+            # Read from HDUList
+            hdulist = fits.open(tmp_file.name)
+            spec = Spectrum1D.read(hdulist, format="SDSS-III/IV spec")
+            assert isinstance(spec, Spectrum1D)
+            assert spec.flux.size > 0
+
 
 @pytest.mark.remote_data
 def test_sdss_spspec():
     sp_pattern = 'spSpec-51957-0273-016.fit.'
+    with urllib.request.urlopen('http://das.sdss.org/spectro/1d_26/0273/1d/spSpec-51957-0273-016.fit') as response:
```

Comment:
Maybe worth adding a quick comment here clarifying that these two cases are for the file-like object and also separately "load a file from disk"?  It's not super-obvious on a quick read.

Reply:
My comments there were rather terse, added a few words to hopefully make the different cases more clear